apps/namadillo/src/App/Swap/SwapCalculations.tsx

```diff
@@ -0,0 +1,353 @@
+import { ActionButton, Stack } from "@namada/components";
+import { mapUndefined } from "@namada/utils";
+import { ConnectProviderButton } from "App/Common/ConnectProviderButton";
+import { SelectAssetModal } from "App/Common/SelectAssetModal";
+import { SelectWalletModal } from "App/Common/SelectWalletModal";
+import { TransactionFeeButton } from "App/Common/TransactionFeeButton";
+import { SwapArrowsIcon } from "App/Icons/SwapArrowsIcon";
+import { defaultShieldedAccountAtom } from "atoms/accounts";
+import {
+  namadaAssetsSortedAtom,
+  namadaShieldedAssetsAtom,
+} from "atoms/balance";
+import { tokenPricesFamily } from "atoms/prices/atoms";
+import BigNumber from "bignumber.js";
+import { TransactionFeeProps, useTransactionFee } from "hooks";
+import { useAvailableAmountMinusFees } from "hooks/useAvailableAmountMinusFee";
+import { useWalletManager } from "hooks/useWalletManager";
+import { wallets } from "integrations";
+import { KeplrWalletManager } from "integrations/Keplr";
+import { useAtom, useAtomValue, useSetAtom } from "jotai";
+import { useCallback, useEffect, useRef, useState } from "react";
+import { NamadaAsset } from "types";
+import { SwapSource } from "./SwapSource";
+import { useSwapSimulation } from "./hooks/useSwapSimulation";
+import { useSwapValidation } from "./hooks/useSwapValidation";
+import { SwapQuote, SwapState, SwapStatus } from "./state";
+import { swapQuoteAtom, swapStateAtom, swapStatusAtom } from "./state/atoms";
+
+const keplr = new KeplrWalletManager();
+
+export const SwapCalculations = (): JSX.Element => {
+  // Local state
+  const [sellAssetSelectorModalOpen, setSellAssetSelectorModalOpen] =
+    useState(false);
+  const [buyAssetSelectorModalOpen, setBuyAssetSelectorModalOpen] =
+    useState(false);
+  const [showConnectToWalletButton, setShowConnectToWalletButton] =
+    useState(false);
+  const [walletSelectorModalOpen, setWalletSelectorModalOpen] = useState(false);
+
+  // Feature state
+  const [swapState, setSwapState] = useAtom(swapStateAtom);
+  const { data: quote } = useAtomValue(swapQuoteAtom);
+  const setStatus = useSetAtom(swapStatusAtom);
+
+  // Global state
+  const sortedAssets = useAtomValue(namadaAssetsSortedAtom);
+  const { data: assetsWithBalance } = useAtomValue(namadaShieldedAssetsAtom);
+  const { data: tokenPrices } = useAtomValue(
+    tokenPricesFamily(sortedAssets.map((a) => a.address))
+  );
+  const feeProps = useTransactionFee(["IbcTransfer"]);
+  const shieldedAccountAddress = useAtomValue(
+    defaultShieldedAccountAtom
+  )?.address;
+
+  useSwapSimulation();
```

Comment:
should we put this in a useEffect with empty dependency array?

Reply:
We can't cause we break react hook rules, I don't think it's that bad tho as we use `useEffect` inside `useSwapSimulation`